pkg/shell/client/shell_client.go

```diff
@@ -136,77 +136,75 @@ func interactiveShell() {
 	}
 	printShellGreeting(console)
 
-	for {
-		// Try to dial the shell.sock. Since it takes a moment for the server to come up and this
-		// is meant for interactive use we'll try to be helpful and retry the dialing until
-		// server comes up.
-		conn, err := dialShell(console)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error: %s\n", err)
-			os.Exit(1)
+	// Try to dial the shell.sock. Since it takes a moment for the server to come up and this
```

Comment:
The loop was there to retry until the agent has created the socket. How about we just loop around `dialShell`? 

OTOH don't really have anything against dropping the loop. Please update the comment though :)